lib/instance.js

```diff
@@ -617,7 +617,7 @@ Instance.prototype.save = function(options) {
           transaction: options.transaction,
           logging: options.logging
         }).then(function () {
-          return self[include.association.accessors.set](instance, {save: false});
+          return self[include.association.accessors.set](instance, {save: false, logging: options.logging});
```

Comment:
Yes, this is an unusual case of a method which is generally async which on this one particular occasion is sync (and so doesn't require `options.logging`.

I added the `options.logging` option here for two reasons:
1. it simplifies the testing shim framework https://github.com/sequelize/sequelize/pull/3843 as then that doesn't have to deal with this special case
2. future-proofing in case things change

Passing the extra option doesn't hurt anyway.

But I can remove this if it's important.